### DISCUSSION TITLE
ToolBar

### DISCUSSION THREAD

**Discussion by PaoloPetra** [iTwin/itwinjs-core#3187] [Q&A] [answered]
Hi good morning, how can I emulate a toolbar inside a widget? similar to the image I am attaching.
I searched through the examples but I don't seem to find anything similar

Thanks for your help

Paul

![toolbarNew](https://user-images.githubusercontent.com/40106886/152972730-828be76c-0eb6-4e3d-9c0e-2fef9f0ea72a.png)

**Comment by bsteinbk** [comment, marked as answer]
@PaoloPetra - You can see a similar toolbar created [here](https://github.com/iTwin/itwinjs-core/blob/53820bf29150af66d28515265f6703e909c5c93c/test-apps/ui-items-providers-test/src/ui/widgets/VisibilityWidget.tsx#L118).  A fairly recent addition to the family of "iTwin" packages that Bentley provides is a standalone UI package[ itwinui-react](https://www.npmjs.com/package/@itwin/itwinui-react) and it provides a ButtonGroup componenent that you may also consider using.  See example [here](https://itwin.github.io/iTwinUI-react/?path=/docs/buttons-buttongroup--with-icons).

**Reply by aruniverse** [reply]
The exact component used in your screenshot can be found [here](https://github.com/iTwin/viewer-components-react/blob/master/packages/itwin/tree-widget/src/components/header/TreeHeader.tsx)
